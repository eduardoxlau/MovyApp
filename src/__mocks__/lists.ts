const lists = [
  {
    name: "Rafael's list",
    movies: [],
  },
];

export default lists;
