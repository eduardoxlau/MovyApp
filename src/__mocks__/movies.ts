const movies = [
  {
    id: 143,
    title: 'International Research Orchestrator',
    overview:
      'Blanditiis similique pariatur provident dolorem repellendus. Accusamus numquam alias in saepe pariatur tenetur. Dolor consequatur quae repudiandae nulla ullam iusto odit repellat qui. Impedit sit facere nostrum nobis dolor. Magnam maiores sunt adipisci excepturi nulla sed aperiam.',
    card_url: 'http://localhost:4000/card1.png',
    poster_url: 'http://localhost:4000/view1.png',
    backdrop_url: 'http://localhost:4000/avatar.png',
    stars: 4,
    genres: [
      {
        name: 'Horror',
      },
      {
        name: 'Family',
      },
    ],
  },
  {
    id: 140,
    title: 'International Brand Representative',
    overview:
      'Iure vel aut exercitationem possimus neque animi. Quo aperiam molestiae consectetur. Soluta exercitationem rem voluptate ipsa ipsum quisquam eum. Aut qui accusamus minus nostrum excepturi et tempore ex eos. Culpa eum perferendis laboriosam ea.',
    card_url: 'http://localhost:4000/card1.png',
    poster_url: 'http://localhost:4000/view1.png',
    backdrop_url: 'http://localhost:4000/avatar.png',
    stars: 2,
    genres: [
      {
        name: 'Fun',
      },
    ],
  },
  {
    id: 147,
    title: 'Future Response Designer',
    overview:
      'Doloribus ducimus quia ex qui quia tenetur aperiam odit ut. Amet repellendus laudantium. Omnis adipisci est ut dolor doloremque et. Sunt consequatur dolorem et dolore rem qui. Voluptatem necessitatibus aperiam.',
    card_url: 'http://localhost:4000/card1.png',
    poster_url: 'http://localhost:4000/view1.png',
    backdrop_url: 'http://localhost:4000/avatar.png',
    stars: 5,
    genres: [
      {
        name: 'Action',
      },
    ],
  },
  {
    id: 134,
    title: 'Future Data Consultant',
    overview:
      'Distinctio accusamus enim optio quis. Dignissimos non autem. Dolores fugit architecto quo aut.',
    card_url: 'http://localhost:4000/card1.png',
    poster_url: 'http://localhost:4000/view4.png',
    backdrop_url: 'http://localhost:4000/avatar.png',
    stars: 3,
    genres: [
      {
        name: 'Fun',
      },
      {
        name: 'Horror',
      },
      {
        name: 'Family',
      },
    ],
  },
  {
    id: 149,
    title: 'Forward Tactics Administrator',
    overview:
      'Ipsam officia et illo. Sit aut illo consequatur rerum ipsum culpa temporibus et laboriosam. Cupiditate aut ipsam modi dolor facilis fugit maiores. Recusandae veniam ad doloremque corrupti fuga molestiae sit placeat. Sint quisquam illo consequatur itaque.',
    card_url: 'http://localhost:4000/more2.png',
    poster_url: 'http://localhost:4000/view2.png',
    backdrop_url: 'http://localhost:4000/matrix.png',
    stars: 1,
    genres: [
      {
        name: 'Fun',
      },
      {
        name: 'Horror',
      },
      {
        name: 'Family',
      },
    ],
  },
  {
    id: 139,
    title: 'Forward Solutions Liaison',
    overview:
      'Quasi est hic ipsa ipsa quod. Incidunt eius maiores aut. Ut nihil molestiae harum exercitationem nisi. Esse repellat pariatur hic porro deleniti iste sint. Sapiente non iste dolores nulla.',
    card_url: 'http://localhost:4000/more4.png',
    poster_url: 'http://localhost:4000/view3.png',
    backdrop_url: 'http://localhost:4000/padrino.jpeg',
    stars: 3,
    genres: [
      {
        name: 'Fun',
      },
      {
        name: 'Horror',
      },
    ],
  },
  {
    id: 144,
    title: 'Dynamic Group Representative',
    overview:
      'Optio assumenda ipsum quos temporibus repudiandae et. Esse doloribus voluptas. Error molestiae error.',
    card_url: 'http://localhost:4000/more4.png',
    poster_url: 'http://localhost:4000/view3.png',
    backdrop_url: 'http://localhost:4000/padrino.jpeg',
    stars: 1,
    genres: [
      {
        name: 'Fun',
      },
      {
        name: 'Horror',
      },
    ],
  },
  {
    id: 142,
    title: 'Dynamic Communications Executive',
    overview:
      'Tempora incidunt et. Officia blanditiis dolorem ex. Est vitae commodi error culpa quas doloremque. Quo dolores suscipit ex velit et tempora.',
    card_url: 'http://localhost:4000/card2.png',
    poster_url: 'http://localhost:4000/view2.png',
    backdrop_url: 'http://localhost:4000/matrix.png',
    stars: 2,
    genres: [
      {
        name: 'Action',
      },
      {
        name: 'Fun',
      },
      {
        name: 'Horror',
      },
      {
        name: 'Family',
      },
    ],
  },
];

export default movies;
