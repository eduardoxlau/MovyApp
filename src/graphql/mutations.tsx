import { gql } from '@apollo/client';

export const LOGIN = gql`
  mutation Login($input: LoginInput!) {
    login(input: $input) {
      access_token
      user {
        id
        email
        full_name
        admin
        photo_path
      }
    }
  }
`;
export const CREATE_USER = gql`
  mutation CreateUser($input: CreateUserInput!) {
    createUser(input: $input) {
      id
      email
      full_name
      admin
      photo_path
    }
  }
`;

export const UPDATE_USER = gql`
  mutation UpdateUser($input: UpdateUserInput!) {
    updateUser(input: $input) {
      status
    }
  }
`;

export const REMOVE_USER = gql`
  mutation DeleteUser {
    deleteUser {
      status
    }
  }
`;

export const CREATE_LIST = gql`
  mutation CreateList($input: CreateListInput!) {
    createList(input: $input) {
      id
      name
      description
    }
  }
`;

export const EDIT_LIST = gql`
  mutation UpdateList($input: UpdateListInput!) {
    updateList(input: $input) {
      status
    }
  }
`;

export const REMOVE_LIST = gql`
  mutation DeleteList($id: Float!) {
    deleteList(id: $id) {
      status
    }
  }
`;

export const REMOVE_MOVIE_LIST = gql`
  mutation RemoveMovieToList($input: ListMovieInput!) {
    removeMovieToList(input: $input) {
      status
    }
  }
`;

export const ADD_MOVIE_LIST = gql`
  mutation AddMovieToList($input: ListMovieInput!) {
    addMovieToList(input: $input) {
      status
    }
  }
`;

export const SEEN_MOVIE = gql`
  mutation seenMovie($movieId: Float!) {
    seenMovie(movieId: $movieId) {
      status
    }
  }
`;
