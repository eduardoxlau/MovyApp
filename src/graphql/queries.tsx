import { gql } from '@apollo/client';

export const GET_USER = gql`
  query GetUser {
    getUser {
      id
      email
      full_name
      admin
      lists {
        description
        movies {
          id
          title
        }
      }
    }
  }
`;

export const GET_MOVIES_SEEN = gql`
  query GetMoviesSeen {
    getMoviesSeen {
      id
      title
      overview
      trailer_url
      card_url
      poster_url
      backdrop_url
      stars
      genres {
        name
      }
    }
  }
`;

export const GET_MOVIE = gql`
  query GetMovie($id: Float!) {
    getMovie(id: $id) {
      id
      title
      overview
      trailer_url
      card_url
      poster_url
      backdrop_url
      stars
      genres {
        name
      }
    }
  }
`;

export const GET_MOVIES = gql`
  query GetMovies {
    getMovies(limit: 20) {
      items {
        id
        title
        overview
        trailer_url
        card_url
        poster_url
        backdrop_url
        stars
        genres {
          name
        }
      }
    }
  }
`;

export const GET_MOVIES_PAGINATE = gql`
  query GetMovies($page: Float, $limit: Float) {
    getMovies(page: $page, limit: $limit) {
      meta {
        totalPages
        currentPage
      }
      items {
        id
        title
        overview
        trailer_url
        card_url
        poster_url
        backdrop_url
        stars
        genres {
          name
        }
      }
    }
  }
`;

export const GET_LISTS = gql`
  query GetLists {
    getLists {
      id
      name
      description
      movies {
        id
        title
        trailer_url
        card_url
      }
    }
  }
`;
