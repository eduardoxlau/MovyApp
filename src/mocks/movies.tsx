import { ItemInterface } from 'components/card';

import Card1 from 'assets/card1.png';
import Card2 from 'assets/card2.png';
import Card3 from 'assets/card3.png';

import View1 from 'assets/view1.png';
import View2 from 'assets/view2.png';
import View3 from 'assets/view3.png';
import View4 from 'assets/view4.png';

import More1 from 'assets/more1.png';
import More2 from 'assets/more2.png';
import More3 from 'assets/more3.png';
import More4 from 'assets/more4.png';

export const moviesSuggestions: ItemInterface[] = [
  {
    id: 0,
    img: More1,
    title: 'The matrix',
    description:
      'It depicts a dystopian "future in which reality as perceived by humans is actually" a simulated reality called the Matrix.',
  },
  {
    id: 1,
    img: More2,
    title: 'The matrix',
    description:
      'It depicts a dystopian "future in which reality as perceived by humans is actually" a simulated reality called the Matrix.',
  },
  {
    id: 2,
    img: More3,
    title: 'The matrix',
    description:
      'It depicts a dystopian "future in which reality as perceived by humans is actually" a simulated reality called the Matrix.',
  },
  {
    id: 3,
    img: More4,
    title: 'The matrix',
    description:
      'It depicts a dystopian "future in which reality as perceived by humans is actually" a simulated reality called the Matrix.',
  },
];

export const movies: ItemInterface[] = [
  {
    id: 0,
    title: 'The matrix',
    img: Card2,
    description:
      'During a manned mission to Mars, Astronaut Mark Watney is presumed dead after a fierce storm and left behind by his crew.',
  },
  {
    id: 1,
    title: 'The Shawshank Redemption ',
    img: Card3,
    description:
      'During a manned mission to Mars, Astronaut Mark Watney is presumed dead after a fierce storm and left behind by his crew.',
  },
  {
    id: 2,
    title: 'El Padrino',
    img: Card1,
    description:
      'During a manned mission to Mars, Astronaut Mark Watney is presumed dead after a fierce storm and left behind by his crew.',
  },
  {
    id: 3,
    title: 'The matrix',
    img: Card2,
    description:
      'During a manned mission to Mars, Astronaut Mark Watney is presumed dead after a fierce storm and left behind by his crew.',
  },
  {
    id: 4,
    title: 'The Shawshank Redemption ',
    img: Card3,
    description:
      'During a manned mission to Mars, Astronaut Mark Watney is presumed dead after a fierce storm and left behind by his crew.',
  },
  {
    id: 5,
    title: 'El Padrino',
    img: Card1,
    description:
      'During a manned mission to Mars, Astronaut Mark Watney is presumed dead after a fierce storm and left behind by his crew.',
  },
];

export const moviesByMovy: ItemInterface[] = [
  {
    id: 0,
    title: 'The Dark Knight',
    img: View1,
    description:
      'During a manned mission to Mars, Astronaut Mark Watney is presumed dead after a fierce storm and left behind by his crew.',
  },
  {
    id: 1,
    title: '12 Angry Men',
    img: View2,
    description:
      'During a manned mission to Mars, Astronaut Mark Watney is presumed dead after a fierce storm and left behind by his crew.',
  },
  {
    id: 2,
    title: 'La lista de Schindler',
    img: View3,
    description:
      'During a manned mission to Mars, Astronaut Mark Watney is presumed dead after a fierce storm and left behind by his crew.',
  },
  {
    id: 3,
    title: ' Pulp Fiction',
    img: View4,
    description:
      'During a manned mission to Mars, Astronaut Mark Watney is presumed dead after a fierce storm and left behind by his crew.',
  },
  {
    id: 4,
    title: '12 Angry Men',
    img: View2,
    description:
      'During a manned mission to Mars, Astronaut Mark Watney is presumed dead after a fierce storm and left behind by his crew.',
  },
  {
    id: 5,
    title: 'La lista de Schindler',
    img: View3,
    description:
      'During a manned mission to Mars, Astronaut Mark Watney is presumed dead after a fierce storm and left behind by his crew.',
  },
  {
    id: 6,
    title: ' Pulp Fiction',
    img: View4,
    description:
      'During a manned mission to Mars, Astronaut Mark Watney is presumed dead after a fierce storm and left behind by his crew.',
  },
];
