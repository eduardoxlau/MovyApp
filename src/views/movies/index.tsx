import { useState } from 'react';
import { useQuery } from '@apollo/client';

import { GET_MOVIES_PAGINATE } from 'graphql/queries';
import { ItemInterface } from 'components/card';

const Movies = () => {
  const [page, setPage] = useState(1);
  let movies: ItemInterface[] = [];
  let pages = 1;
  const { data } = useQuery(GET_MOVIES_PAGINATE, {
    variables: {
      page,
      limit: 8,
    },
  });

  if (data) {
    pages = data.getMovies.meta.totalPages;
    movies = data.getMovies.items;
  }

  return (
    <div className="bg-black flex flex-col flex-grow">
      <div className="container mx-auto mt-40 mb-20">
        <div className="flex w-full flex-col md:flex-row text-white flex-wrap">
          {movies?.map((movie: ItemInterface) => (
            <div className="flex flex-col px-3 w-full md:w-1/4 mb-10">
              <img className="h-50 w-full" src={movie.card_url} alt="" />
              <div className="flex my-4 items-center">
                <div className="text-xl">{movie.title}</div>
                <div className="rounded-full border border-white ml-auto text-lg px-4 py-0 font-bold opacity-90">
                  PG13
                </div>
              </div>
              <div>{`${movie.overview.slice(0, 100)} ...`}</div>
            </div>
          ))}
        </div>
        <div className="flex justify-center text-center text-white">
          {Array(pages)
            .fill('')
            .map((_, index) => (
              <div
                className={`m-1 px-2 py-1 rounded cursor-pointer ${
                  page === index + 1 && 'bg-blue-500'
                }`}
                onClick={() => setPage(index + 1)}
              >
                {index + 1}
              </div>
            ))}
        </div>
      </div>
    </div>
  );
};

export default Movies;
