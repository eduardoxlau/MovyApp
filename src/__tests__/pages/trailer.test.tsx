/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */
import wait from 'waait';
import { createMemoryHistory } from 'history';
import { Router, Route } from 'react-router-dom';
import { MockedProvider } from '@apollo/client/testing';
import { render, fireEvent } from '@testing-library/react';

import {
  REMOVE_MOVIE_LIST,
  ADD_MOVIE_LIST,
  SEEN_MOVIE,
} from 'graphql/mutations';
import movies from '__mocks__/movies.mock';
import lists from '__mocks__/lists.mock';
import { GET_MOVIE, GET_LISTS, GET_MOVIES_SEEN } from 'graphql/queries';
import Trailer from 'views/trailer';

let queryCalled = false;

const mocks = [
  {
    request: {
      query: REMOVE_MOVIE_LIST,
    },
    result: { data: { status: 'success' } },
  },
  {
    request: {
      query: ADD_MOVIE_LIST,
      variables: { input: { listId: 8, movieId: 143 } },
    },
    result: { data: { status: 'success' } },
  },
  {
    request: {
      query: SEEN_MOVIE,
      variables: { movieId: 143 },
    },
    result: {
      data: { status: 'success' },
    },
  },
  {
    request: {
      query: GET_MOVIE,
      variables: {
        id: 1,
      },
    },
    result: { data: { getMovie: movies[0] } },
  },
  {
    request: {
      query: GET_LISTS,
    },
    newData: () => {
      if (queryCalled) {
        const newList = lists;
        newList[0].movies = [movies[0]];
        return { data: { getLists: newList } };
      }
      return { data: { getLists: lists } };
    },
  },
  {
    request: {
      query: GET_MOVIES_SEEN,
    },
    result: { data: { getMoviesSeen: movies } },
  },
];

describe('<Home />', () => {
  const history = createMemoryHistory({ initialEntries: ['/trailer/1'] });

  it('should render snapshot Home', async () => {
    const { asFragment } = render(
      <MockedProvider
        mocks={mocks}
        defaultOptions={{ watchQuery: { fetchPolicy: 'no-cache' } }}
      >
        <Router history={history}>
          <Route path="/trailer/:id">
            <Trailer />
          </Route>
        </Router>
      </MockedProvider>
    );
    await wait(0);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should allow watch video iframe', async () => {
    const { getByTestId } = render(
      <MockedProvider
        mocks={mocks}
        defaultOptions={{ watchQuery: { fetchPolicy: 'no-cache' } }}
      >
        <Router history={history}>
          <Route path="/trailer/:id">
            <Trailer />
          </Route>
        </Router>
      </MockedProvider>
    );
    await wait(0);
    const video = getByTestId('play-video');
    fireEvent.click(video);
    await wait(200);
    expect(video).not.toBeVisible();
  });
  it('should add movie to list', async () => {
    const { getAllByTestId, getByText } = render(
      <MockedProvider
        mocks={mocks}
        defaultOptions={{ watchQuery: { fetchPolicy: 'no-cache' } }}
      >
        <Router history={history}>
          <Route path="/trailer/:id">
            <Trailer />
          </Route>
        </Router>
      </MockedProvider>
    );
    await wait(0);
    const movie = getAllByTestId('user-list')[0];
    fireEvent.click(movie);
    queryCalled = true;
    await wait(200);
    const removeButton = getByText('Remove to list');
    expect(removeButton).toBeVisible();
  });
});
