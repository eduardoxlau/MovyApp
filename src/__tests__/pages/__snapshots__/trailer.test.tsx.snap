// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render snapshot Home 1`] = `
<DocumentFragment>
  <div
    class="bg-black flex flex-col flex-grow text-white"
  >
    <div
      class="container mx-auto my-40"
    >
      <div
        class="flex mb-4 flex-col md:flex-row"
      >
        <div
          class="flex"
        >
          <div>
            2019 . Action, Thiller
          </div>
          <div
            class="flex items-center ml-20"
          >
            <img
              alt=""
              class="w-5 mr-3"
              src="save.png"
            />
            Watch Later
          </div>
        </div>
        <div
          class="flex items-center mt-3 md:mt-0 md:ml-20 relative  "
        >
          <div>
            <div
              class="flex cursor-pointer"
            >
              <img
                alt=""
                class="w-7 h-7 mr-3"
                src="add.png"
              />
              add to list
            </div>
            <div
              class="flex flex-col bg-blue-500 p-3 rounded top-8 min-w-full w-max text-center hidden"
            >
              <div
                class="cursor-pointer hover:font-bold"
                data-testid="user-list"
              >
                Rafael's list
              </div>
              <div
                class="cursor-pointer font-bold  text-lg"
              >
                Create new List
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="text-6xl font-black "
      >
        International Research Orchestrator
      </div>
      <div
        class="flex mt-3 mb-16 items-center justify-around md:justify-start"
      >
        <div
          class="bg-pink-600 px-2 py-0.5 rounded md:mr-5 text-sm flex items-center"
        >
          2.30h
        </div>
        <div
          class="mx-10 md:mr-6 text-lg flex items-center text-center flex-col md:flex-row"
        >
          <img
            alt=""
            class="w-6 mr-3"
            src="camera.png"
          />
          <div>
            Quentien cuarentino
          </div>
        </div>
        <div
          class="flex items-center"
        >
          <img
            alt=""
            class="w-4 mr-1"
            src="star.png"
          />
          <img
            alt=""
            class="w-4 mr-1"
            src="star.png"
          />
          <img
            alt=""
            class="w-4 mr-1"
            src="star.png"
          />
          <img
            alt=""
            class="w-4 mr-1"
            src="star.png"
          />
        </div>
      </div>
      <div
        class="w-2/3 overflow-hidden relative m-auto "
        style="height: 500px;"
      >
        <iframe
          class="w-full h-full absolute"
          height="auto"
          src="https://www.youtube.com/embed/hf_wom7OiW8"
          width="100%"
        />
        <div
          class="absolute w-full h-full bg-black opacity-50"
        />
        <div
          class="absolute w-full h-full  flex justify-center items-center opacity-80"
        >
          <img
            alt=""
            class="w-40 cursor-pointer"
            data-testid="play-video"
            src="bt_play.png"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
